src/main/java/org/opensearch/ad/rest/RestExecuteAnomalyDetectorAction.java

```diff
@@ -115,21 +146,36 @@ private String validateAdExecutionInput(AnomalyDetectorExecutionInput input) {
     }
 
     @Override
-    public List<Route> routes() {
-        return ImmutableList.of();
-    }
-
-    @Override
-    public List<ReplacedRoute> replacedRoutes() {
+    public List<ReplacedRouteHandler> replacedRouteHandlers() {
         return ImmutableList
             .of(
                 // get AD result, for regular run
-                new ReplacedRoute(
+                new ReplacedRouteHandler(
                     RestRequest.Method.POST,
-                    String.format(Locale.ROOT, "%s/{%s}/%s", AnomalyDetectorPlugin.AD_BASE_DETECTORS_URI, DETECTOR_ID, RUN),
+                    String.format(Locale.ROOT, "%s/{%s}/%s", AnomalyDetectorExtension.AD_BASE_DETECTORS_URI, DETECTOR_ID, RUN),
                     RestRequest.Method.POST,
-                    String.format(Locale.ROOT, "%s/{%s}/%s", AnomalyDetectorPlugin.LEGACY_OPENDISTRO_AD_BASE_URI, DETECTOR_ID, RUN)
+                    String.format(Locale.ROOT, "%s/{%s}/%s", AnomalyDetectorExtension.LEGACY_OPENDISTRO_AD_BASE_URI, DETECTOR_ID, RUN),
+                    handleRequest
                 )
             );
     }
+
+    private ExtensionRestResponse getExecuteDetectorResponse(
+        RestStatus restStatus,
+        RestRequest request,
+        AnomalyResultResponse response,
+        String content
+    ) throws IOException {
+        ExtensionRestResponse extensionRestResponse;
+        if (content.isEmpty()) {
```

Comment:
just curious, what's the point of checking if the content is empty if we're only using this method once and passing "" for the content.

Reply:
The reason I have added a content parameter is because as per the previous code written for plugins, if we find the error in validating request then it throws a error after setting it in ByteResponse. So from wherever this api would have been called different type of error handling would be done. I have passed "" placeholder for content when the api execution is done without any error.

Reply:
Ah I missed that first invocation of the method after checking for errors. LGTM

Reply:
We don't need an extra check here for content. If there's any error in the response the code execution would go to the catch block [here](https://github.com/opensearch-project/anomaly-detection/pull/890/files#diff-de43e12dd14419a5047d6b20decd5f99683bf094c7d0331f0cebbe08a74b92f8R116). Also, we need to handle the error by checking the correct exception. Passing `""` is not the correct way to do so. I would say let's handle the exact error throwing in the future not for this API but for all the other we have so far and create an issue for the same.

Reply:
So reduce the diff in implementation and send the right error to the user, I have added a piece of [code](https://github.com/opensearch-project/anomaly-detection/pull/890/files#diff-de43e12dd14419a5047d6b20decd5f99683bf094c7d0331f0cebbe08a74b92f8R86) here.  This code snippet replicate the [earlier implementation](https://github.com/opensearch-project/anomaly-detection/pull/890/files#diff-de43e12dd14419a5047d6b20decd5f99683bf094c7d0331f0cebbe08a74b92f8L76) in returning the response. 

I am returning "" [here](https://github.com/opensearch-project/anomaly-detection/pull/890/files#diff-de43e12dd14419a5047d6b20decd5f99683bf094c7d0331f0cebbe08a74b92f8R108)  because I have created a common method [below](https://github.com/opensearch-project/anomaly-detection/pull/890/files#diff-de43e12dd14419a5047d6b20decd5f99683bf094c7d0331f0cebbe08a74b92f8R163) by following dry principle as I am using the same piece of code at 2 places. That is reason why I have put a check of content.isEmpty in the method.

Reply:
You can check for 
1. the RestStatus if it's OK or a BAD_REQUEST
2. response since you are already passing `null` for error. 
I would prefer the check for RestStatus.  For the successful response you can pass the content as `response.toXContent(JsonXContent.contentBuilder(), ToXContent.EMPTY_PARAMS)` instead of "".

Reply:
cool